odl/tomo/geometry/conebeam.py

```diff
@@ -1236,6 +1238,178 @@ def frommatrix(cls, apart, dpart, src_radius, det_radius, init_matrix):
+
+    Parameters
+    ----------
+    space : `DiscreteLp`
+        Reconstruction space, the space of the volumetric data to be
+        projected. Must be 2-dimensional.
+    src_radius : nonnegative float
+        Radius of the source circle.
+    det_radius : nonnegative float
+        Radius of the detector circle.
+    num_angles : int, optional
+        Number of angles.
+        Default: Enough to fully sample the data, see Notes.
+    det_shape : int or sequence of int, optional
+        Number of detector pixels.
+        Default: Enough to fully sample the data, see Notes.
+
+    Returns
+    -------
+    geometry : `DivergentBeamGeometry`
+        If ``space`` is 2d, return a `FanFlatGeometry`.
+        If ``space`` is 3d, return a `CircularConeFlatGeometry`.
+
+    Examples
+    --------
+    Create geometry from 2d space and check the number of data points:
+
+    >>> space = odl.uniform_discr([-1, -1], [1, 1], (20, 20))
+    >>> geometry = cone_beam_geometry(space, src_radius=5, det_radius=5)
+    >>> geometry.angles.size
+    77
+    >>> geometry.detector.size
+    28
+
+    If the source is close to the object, the detector becomes larger due
+    to more magnification:
+
+    >>> geometry = cone_beam_geometry(space, src_radius=3, det_radius=9)
+    >>> geometry.angles.size
+    79
+    >>> geometry.detector.size
+    52
+
+    Notes
+    -----
+    According to `Mathematical Methods in Image Reconstruction`_
+    (page 75--76), for a function :math:`f : \\mathbb{R}^2 \\to \\mathbb{R}`
+    that has compact support
+
+    .. math::
+        \| x \| > \\rho  \implies f(x) = 0,
+
+    and is essentially bandlimited
+
+    .. math::
+       \| \\xi \| > \\Omega \implies \\hat{f}(\\xi) \\approx 0,
+
+    then, in order to fully reconstruct the function from a fan beam ray
+    transform with source-detector distance :math:`r` (assuming all detector
+    points have the same distance to the source), the function should be
+    sampled at an angular interval :math:`\\Delta \psi` such that
+
+    .. math::
+        \\Delta \psi \leq \\frac{r + \\rho}{r}\, \\frac{\\pi}{\\rho \\Omega},
+
+    and the detector should be sampled with an angular interval
+    :math:`\\Delta \\alpha` that satisfies
+
+    .. math::
+        \\Delta \\alpha \leq \\frac{\\pi}{r \\Omega}.
+
+    For a flat detector, the angular interval is smallest in the center
+    of the fan and largest at the boundaries. The worst-case relation
+    between the linear and angular sampling intervals are
+
+    .. math::
+        \\Delta s = R \\Delta \\alpha, \quad R^2 = r^2 + (w / 2)^2,
+
+    where :math:`w` is the width of the detector.
+    Thus, to satisfy the angular detector condition one can choose
+
+    .. math::
+        \\Delta s \leq \\frac{\\pi \sqrt{r^2 + (w / 2)^2}}{r \\Omega}.
+
+    The geometry returned by this function satisfies these conditions exactly.
+
+    If the domain is 3-dimensional, a circular cone beam geometry is
```

Comment:
We never say that explicitly anywhere, nor do we mention how the angles are chosen. I'll add that.